project/frontend/src/components/Authentication.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+
+/**
+ * A context containing the following fields:
+ * - currentUser {!GoogleUser}
+ * - signIn {function(): undefined}
+ * - signOut {function(): undefined}
+ *
+ * Components that need authentication information may listen to this
+ * context using the useContext(AuthContext) hook, AuthContext.Consumer
+ * component, or the ClassWidget.contextType property.
+ */
+const AuthContext = React.createContext({});
+
+/**
+ * A high-order component responsible for initializing and maintaining authentication state
+ * within the application. Serves the AuthContext to child components.
+ */
+function Authentication(props) {
+  const [GoogleAuth, setGoogleAuth] = useState({
+    signIn: () => {},
+    signOut: () => {},
+  });
+  const [currentUser, setCurrentUser] = useState({ get: undefined });
```

Comment:
What does the `get` do?

Also let's make this null instead of undefined.

Reply:
The get doesn't do anything. The benefit here is that by wrapping the GoogleUser object inside an object React recognizes the user as a new object and actually rerenders the context state. I just chose the property name get because it makes sense in the context of calling `authContext.currentUser.get` instead of something redundant like `authContext.currentUser.user`.

Reply:
Gotcha, thanks for explaining!